### DISCUSSION TITLE
Is it worth adding docs about sharing styles with CSS Module Scripts

### DISCUSSION THREAD

**Discussion by andyjessop** [lit/lit#4522] [General]
I'm struggling to find much info on browser support, but as I understand it, you could share CSS as actual stylesheets and just adopt them as needed into elements.

Should there be docs in the styling section?

**Comment by augustjk** [comment]
There's a bit of confusion because Chrome was shipping it for a while but using the older version of import attributes (formerly import assertions) using syntax
```js
import styleSheet from "./styles.css" assert { type: "css" };
```

I believe Chrome is working on the new import attribute syntax using `with`
```js
import styleSheet from "./styles.css" with { type: "css" };
```
but not shipped yet, or might be behind a flag.

Safari has implemented import attributes, but not CSS module scripts. They shipped import attributes with JSON module scripts only.

Tools like rollup or esbuild have mixed support for import attributes. I think there exist some plugins but unsure. Some exist to simply allow the parser to not break with importing CSS, whereas some do a conversion to JS.

It would be good to track this better with links to issues. But documenting it might be a bit premature.

**Reply by andyjessop** [reply]
That's probably why I'm struggling to find much info on it. Thanks very much for the context.

**Reply by augustjk** [reply]
Dumping some related issues here. They're a mix of import attribute support and specifically css module support:

https://github.com/microsoft/TypeScript/issues/46689
https://github.com/evanw/esbuild/issues/3384
https://github.com/rollup/rollup/issues/4818
https://github.com/vitejs/vite/issues/14674
